app/services/ApprovalService.scala

```diff
@@ -41,26 +41,36 @@ class ApprovalService @Inject() (repository: ApprovalRepository) {
       }
     }
 
-    def createApprovalProcess(process: Process): ApprovalProcess = {
-      ApprovalProcess(process.meta.id, ApprovalProcessMeta(process.meta.id, process.meta.title), jsonProcess)
+    def saveReview(approvalProcessReview: ApprovalProcessReview): Future[RequestOutcome[String]] = {
+      reviewRepository.save(approvalProcessReview) map {
+        case Right(_) => Right(approvalProcessReview.ocelotId)
+        case _ => Left(Errors(InternalServiceError))
+      }
     }
 
-    jsonProcess.validate[Process] match {
-      case JsSuccess(process, _) =>
-        saveProcess(createApprovalProcess(process))
-      case JsError(errors) =>
-        logger.error(s"Parsing process failed with the following error(s): $errors")
-        Future { Left(Errors(BadRequestError)) }
+    validateProcess(jsonProcess) match {
+      case Right(process) =>
+        saveProcess(createApprovalProcess(process.meta.id, process.meta.title, initialStatus, jsonProcess)) flatMap {
+          case Right(savedId) =>
+            repository.getById(savedId) flatMap {
+              case Right(approvalProcess) => saveReview(createApprovalProcessReview(process, reviewType, approvalProcess.version))
+              case Left(Errors(NotFoundError :: Nil)) => Future.successful(Left(Errors(NotFoundError)))
+              case Left(_) => Future.successful(Left(Errors(InternalServiceError)))
+            }
+          case errors => Future.successful(errors)
+        }
+      case Left(_) =>
+        Future.successful(Left(Errors(BadRequestError)))
     }
 
   }
 
   def getById(id: String): Future[RequestOutcome[JsObject]] = {
 
     repository.getById(id) map {
-      case error @ Left(Errors(NotFoundError :: Nil)) => error
+      case error @ Left(Errors(NotFoundError :: Nil)) => Left(Errors(NotFoundError))
```

Comment:
Can remove the error @, its not being used

Reply:
will do